src/VecSim/vec_sim.cpp

```diff
@@ -45,6 +45,13 @@ extern "C" int VecSimIndex_DeleteVector(VecSimIndex *index, size_t id) {
 extern "C" double VecSimIndex_GetDistanceFrom(VecSimIndex *index, size_t id, const void *blob) {
     return index->getDistanceFrom(id, blob);
 }
+extern "C" double VecSimIndex_GetDistanceFrom_withFlag(VecSimIndex *index, size_t id,
+                                                       const void *blob, bool normalize) {
+    return index->getDistanceFrom(id, blob, normalize);
+}
+extern "C" void VecSimIndex_PrepareVector(VecSimIndex *index, const void *source, void *dest) {
+    index->prepareVector(source, dest);
+}
```

Comment:
I don't like the name `PrepareVector`... I think that it is better to just call the child at it's name - `NormalizeVector`.
Also, I don't think that this function should be a member of `VecSimIndex`, as it is more of a "static" function. Then the signature may become something like:
 `VecSim_NormalizeVector(const void *source, void *dest, size_t dim, VecSimType type)`
@DvirDukhan WDYT?

Reply:
the idea was to use this function before using `getDistanceFrom_withFlag` without a need to check what type of space we have.
sometimes this function will not normalize so calling it like that will be misleading.